### ISSUE TITLE
script does not upgrade

### ISSUE THREAD

**Issue by ghost** [osTicket/osTicket-1.7#426]
i should be able to copy all files over and go to /scp/upgrade.php   and it sould update it a utomatically all it says is upgrading smile then the little box goes away and doesnt say upgraded because it does not let me do anything until the upgrade is complete.

I am just upgrading from 1.7rc3 to 1.7rc4

**Comment by protich** [comment]
Jason - Any reason why you closed the issue? If you resolved it - then post feedback so others may benefit.

**Comment by ghost** [comment]
Well i wanted to make sure it wasnt just me, doing something wrong,   I have upgraded alot of scripts and throughout the whole development of testing 1.7 i have not been able to upgrade sucessfully one time,  it always acts like its going to do it but doesnt,  and then i just copied all the files  over because its not being used for inproduction and instead of upgrading i have to do a clean install again...

Either i am doing something totally wrong or upgrading 1.7 is a pain,   upgrading should be so easy  but not with 1,7 for some reason,  I have upgraded previous versions like 1.6 and things always went ok

**Comment by protich** [comment]
As far as I know you're the only one reporting upgrade issues. It might make sense to track down the possible culprit.

Do you get any errors in Admin Panel > Dashboard > System logs?

**Comment by ghost** [comment]
nope i never have gotten an error,  it is almost like its always stuck in a look, it will say upgrading then a couple seconds later just quits and never completes the upgrade for me,   then again their always seems to be issue related to godaddy hosting...

It lets me setup and do a clean install just fine,  but when it comes to upgrading for me its next to impossible,  i download all the zip files , extract them to desktop, then i upload all contents of folder then go to example.com/support/scp/upgrade.php to start the upgrade  process...

since I have started testing through the development of 1.7 i have never been able to upgrade,  I always have to just create a new SQL database and do a fresh install,  i dunno if its just me or my hosting or what,  because as you mentioned I am really the only one having an upgrade issue,  no other person who is testing 1.7 seems to be experiencing this

**Comment by protich** [comment]
I'll be interested to get to the bottom of the issue. If possible setup a temp RC3 installation then stage RC4 upgrade and give me admin access. You can email info to peter @ osticket.com. Thanks.

**Comment by ghost** [comment]
it could just be simular to same problem i am having with attachments,  i did a clean fresh install and i am still able only able to upload 1 file even tho i have selected i can upload up to 7 files because im really the only one experiencing  that issue also.

But other then that everything seems to be working for me,  im currently testing 1.7 rc4 out to see if i run into any type of issues.

**Comment by protich** [comment]
I'll debug both issues when I get access. 

What browser (& version) are you using?

**Comment by ghost** [comment]
Im using all the latest browser,  Firefox 17.01  opera 12.12, google chrome and safari

**Comment by ghost** [comment]
Update:

For anyone who is having an issue with multi  file upload on their  osticket installation,  I have been working with peter to try to resolve this issue,   together with his help we have found a fix,   this fix is more so for customers who use godaddy as their hosting provider,  but if your experiencing the same issue with a different hosting provider this fix will work also....

By default godaddy has this setting

cgi.fix_pathinfo = 0

if you don't have a php5.ini file already you will need to create one and upload it to your root directory

and replace cgi.fix_pathinfo = 0   and change it to cgi.fix_pathinfo = 1

then login to your control center panel and click on system process and then click on end web,  this will restart your apache server if you are on shared hosting...

If you are using Windows IIS hosting, go to your hosting control panel and select from the top menu: Content->IIS Management.

Click “Recycle App Pool”. This will do the same as “End web” on Linux.

**Comment by Richiricheh** [comment]
This is fantastic information. I see this as very fitting for the MODS section in the forum. Perhaps Peter see's this information as fitting into a different spot but I see what is being discussed here as a MOD in order to allow OS Ticket to work.

**Comment by ghost** [comment]
is 1.7St on schedule to be released monday march 4th,  was just wondering cause you mentioned on osticket it would be released in 2 weeks,  and you hadnt worked on your pull requests you created or started putting the finishing touches on it yet.

**Comment by support4net** [comment]
Hi,
I've got in the same upgrade errors for several times Is reporting with and odd message that something went wrong and bye bye....
I've work for 2 days making this upgrade process going forward from 1.6ST to .1.7.1, 1.7.3, 1.8rc2 and same story.
It seems that each time is messing up with the DB and then fails to finish the process.
I've tried with a server running on OSE 13.1, with PHP5.4, then returning to an older server with OSE 11.4 (32bit) , PhP 5.3. Same story there and it seems impossible to upgrade from 1.6ST to any other version collected from official ZIP or from GIT.

Any idea if someone managed to find out why is this upgrader process crashing ?
For me is critical because I have a large number of tickets and I would like to migrate to a newer version....

**Comment by greezybacon** [comment]
So we test upgrades from a well used 1.6 help desk up through 1.8. When the upgrade crashes, do you get failure emails to the Admin and/or failure logs in the syslog table?

If you upgrade is critical, we do offer professional support...

**Comment by support4net** [comment]
The upgrades are clear for me that do not work according to the documentation provided. The old 1.6 is stable and work well, while the upgrade to any other version fails in the same way while altering the DB structure.
The upgrades were performed on several servers now on different versions of OS and PHP with the same results.
Our conclusion is that the upgrade script is unreliable and is not performing well.
After each upgrade failure the email were showing errors for calling various missing tables meaning that the DB was altered wrong over several Mysql versions present in Opensuse 11.4, 12.1, 12.2, 13.1.

My upgrade is not critical as we can continue to use the old 1.6 but I'm considering migrating toward other ticketing software as the reported errors are too many on OSE 13.1 with PHP 5.4.....

**Comment by protich** [comment]
@support4net - While we cannot rule out upgrader bugs due to obscure server settings  - I can assure you that thousands of osTicket installations are upgrading just fine. Without access to your server and installation, it's hard for us to guess what could be causing the issue in your case. 

Good luck  with the migration.

**Comment by greezybacon** [comment]
We run osTicket on php 5.4. I don't think that's the issue. You've made it clear that you have upgrade errors. Unfortunately, my magic wand is out on loan. If you have specific error messages you care to post, I can correct the issues.

Unfortunately, migrating to another help desk software is beyond the scope of this project.

**Comment by support4net** [comment]
@protich  I agree with you. There's no obscure settings on my server. It is a fresh installation of OpenSuse 13.1 with nothing special configured. It's a standard setup of PHP and a VirtualHost pointing to /home/osticket  .
By simply copy the old 1.6  to this current fresh server, the app runs well (except the strict errors in apache log).
I think anyone could test this as the environment is clean and takes max. 1h to setup.

**Comment by greezybacon** [comment]
You are a fool. Neither we nor anyone outside your organization has access to your osTicket 1.6 data. How can anyone test it?! How can you assume that I can or should already have guessed what data you have in your database, or that any of the changes made to the software over the last four years would have some adverse affect on your specific data?

This error has likely nothing to do with your server setup, hosting option, php version, or MySQL configuration. More likely, there is some inconsistency in your data, like a charset or collation setting for a table, a mod you have integrated, or any other obscurity that we cannot predict.

If you are unwilling to post a specific error message from the upgrader, please have the decency to close the issue.

**Comment by protich** [comment]
@support4net - Starting with 1.7+, ajax calls require `path_info` enabled. I suspect it's the same issue with your PHP setup/installation.

**Comment by support4net** [comment]
@protich the path_info = 1 is already set.

**Comment by support4net** [comment]
@greezybacon  I'm not a fool and I don't consider you stupid. I simply don't want to provide  the true hostname for a potential PHP remote exploits over a production servers considering the STRICT erros logged into apache. First fix that issues then we will talk about how secure is this app....

I agree we you that it might be a problem with the database structure, but I don't see why it runs smoothly on ver. 1.6...

**Comment by protich** [comment]
@support4net - Post the said error messages you're getting or even apache logs... (fire up FireBug or equivalent and post the error codes..etc). I don't buy your argument about apache logs - no one is asking you to post the entire file.

Without logs or access, it's impossible for us to help -- migrating to a "better" system might be the best course of action for you.

**Comment by support4net** [comment]
@protich you are right and I'll return with the logs tomorrow. I didn't say I'll not post the logs.....
IF you don't buy my arguments then send me your host name and I'll check with a few exploits your security over the night :))) Just kidding but I've seen too many hacked servers and I don't take the risks on the production servers.
